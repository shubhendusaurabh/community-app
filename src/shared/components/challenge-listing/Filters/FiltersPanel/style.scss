@import '~styles/tc-styles';
$panel-space-7: $base-unit + 2;
$panel-space-10: $base-unit * 2;
$panel-space-15: $base-unit * 3;
$panel-space-20: $base-unit * 4;
$panel-space-25: $base-unit * 5;
$panel-space-30: $base-unit * 6;
$panel-space-50: $base-unit * 10;
$panel-space-100: $base-unit * 20;
$panel-space-110: $base-unit * 22;
$panel-radius-1: $corner-radius / 2;
$panel-radius-4: $corner-radius * 2;

.FiltersPanel {
  background: white;
  padding: $panel-space-15 $panel-space-20;
  width: 100%;
  position: relative;

  @include xxs-to-sm {
    position: fixed;
    z-index: 10;
    width: auto;
    height: auto;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    background: $tc-white;
    padding: 15px;
    overflow-y: scroll;
  }

  &.hidden {
    display: none;
  }

  .header {
    display: none;

    @include xxs-to-sm {
      display: flex;
      justify-content: space-between;
      padding-bottom: 25px;
    }
  }

  .title {
    display: none;

    @include xxs-to-sm {
      display: block;
      font-weight: 500;
      font-size: 12px;
      color: $tc-gray-50;
      letter-spacing: 0;
      line-height: $base-unit * 4;
      text-transform: uppercase;
    }
  }

  .close-icon {
    display: none;

    @include xxs-to-sm {
      display: block;
      cursor: pointer;
    }
  }

  .buttons {
    border-left: 1px solid $tc-gray-20;
    float: right;
    padding-left: $panel-space-20;
    position: absolute;
    right: $panel-space-20;
    top: $panel-space-15;

    @include xxs-to-sm {
      float: none;
      display: flex;
      justify-content: center;
      position: relative;
      right: 0;
      top: $panel-space-30;
      width: 100%;
      padding-left: 0;
      margin-bottom: $panel-space-100;
      border-left: none;
    }

    button {
      display: block;
      text-transform: none;
      width: $panel-space-100 - 3;
      margin-left: 0;

      @include xxs-to-sm {
        display: inline-block;
      }

      &.blue {
        color: $tc-white;

        @include xxs-to-sm {
          margin: 0 3px;
        }
      }

      &.white {
        color: $tc-gray-70;
        margin-bottom: $panel-space-15;

        @include xxs-to-sm {
          margin: 0 3px;
        }
      }
    }
  }
}

.filters {
  display: inline-block;
  width: 100%;

  @include xxs-to-sm {
    display: block;
  }

  .filter-row {
    display: flex;
    align-items: center;
    margin-right: $panel-space-100 + 17;

    @include xxs-to-sm {
      display: block;
      margin-right: 0;
    }
  }

  .filter {
    display: flex;
    align-items: center;

    &.keywords {
      width: 100%;
      margin-bottom: $panel-space-15;
      margin-right: $panel-space-15;
    }

    label {
      display: inline-block;
      font-size: 13px;
      line-height: $panel-space-15;
      margin-right: $panel-space-15;
      min-width: 96px;
      text-align: right;

      &.left-label {
        min-width: 58px;
      }

      @include xxs-to-sm {
        text-align: left;

        &.left-label {
          min-width: 96px;
        }
      }
    }

    :global(.Select) {
      display: inline-block;
      width: 100%;
      margin-bottom: 0;

      &#keywordsFilter {
        @include calc(width, '100% - 56px');
      }
    }

    &.track {
      display: flex;
      align-items: center;

      @include calc(width, '100% - 395px');

      margin-right: $panel-space-30;

      @include xxs-to-sm {
        width: initial;
      }
    }

    &.dates {
      display: inline-block;
      text-align: left;

      @include xxs-to-sm {
        margin-top: 20px;
      }
    }
  }
}

.inGroup .filter-row {
  .filter.keywords {
    @include calc(width, '50% - 30px');

    @include xxs-to-sm {
      width: 100%;
    }

    margin-right: $panel-space-30;
  }

  .filter.community {
    @include calc(width, '50% - 30px');

    @include xxs-to-sm {
      width: 100%;
    }

    :global(.Select-value) {
      line-height: 28px;
      top: inherit;
    }

    margin-bottom: $panel-space-15;
    margin-right: $panel-space-30;
  }

  .filter.track {
    @include calc(width, '50% - 30px');

    @include xxs-to-sm {
      width: 100%;
    }
  }
}

// filter by date
.filter.dates {
  :global {
    .DateInput {
      padding: $base-unit - 3;
      line-height: $panel-space-20;
      width: $panel-space-110 + 2;

      input {
        padding: $base-unit;
        height: $panel-space-30;
        margin-bottom: 0;
        font-family: 'Roboto';
        font-weight: 400;
        font-size: 13px;
        color: $tc-black;
        line-height: $panel-space-15 + 1;
      }

      .DateInput__display-text,
      .DateInput__display-text--has-input {
        font-family: 'Roboto';
        font-weight: 400;
        font-size: 13px;
        color: $tc-black;
        line-height: $panel-space-15 + 1;
        padding-left: $panel-space-25;
      }

      .DateInput__display-text {
        position: relative;

        &::before {
          content: '';
          background-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB4PSIwcHgiIHk9IjBweCIgd2lkdGg9IjEycHgiIGhlaWdodD0iMTJweCIgdmlld0JveD0iMCAwIDE2IDE2Ij48ZyA+PHBhdGggZmlsbD0iI0EzQTNBRCIgZD0iTTE1LDJoLTJWMWMwLTAuNi0wLjQtMS0xLTFoLTFjLTAuNiwwLTEsMC40LTEsMXYxSDZWMWMwLTAuNi0wLjQtMS0xLTFINEMzLjQsMCwzLDAuNCwzLDF2MUgxQzAuNCwyLDAsMi40LDAsM3YxMmMwLDAuNiwwLjQsMSwxLDFoMTRjMC42LDAsMS0wLjQsMS0xVjNDMTYsMi40LDE1LjYsMiwxNSwyeiBNMTQsMTRIMlY3aDEyVjE0eiIvPjwvZz48L3N2Zz4=);
          background-position: 0 0;
          background-repeat: no-repeat;
          width: $panel-space-15 - 1;
          height: $panel-space-15 - 1;
          display: inline-block;
          position: absolute;
          left: $panel-space-7;
          top: $panel-space-7 - 1;
        }
      }
    }

    .CalendarMonth table {
      font-weight: 400;
    }

    .CalendarMonth__caption {
      margin-bottom: $panel-space-10;
    }

    .DayPicker__week-header {
      font-weight: 400;
    }

    .DateRangePickerInput__arrow svg {
      width: $panel-space-15;
      height: $panel-space-15;
      fill: $tc-black;
    }

    .DateRangePicker__picker {
      top: $base-unit * 9;
    }

    .DateInput--with-caret::before {
      top: $panel-space-30 + 4;
    }

    .DateInput--with-caret::after {
      top: $panel-space-30 + 5;
    }
  }
}

// filter by keywords
.filter {
  :global {
    .Select-placeholder {
      font-weight: 300;
      font-size: 13px;
      color: $tc-gray-50;
      line-height: $panel-space-30 - 2;
    }

    .Select-input > input:not([type="checkbox"]) {
      height: $panel-space-30 - 2;
      border: none;
      box-shadow: none;
      background: transparent;
      border-radius: 0;
      transition: none;
      padding-left: 0;
      margin-bottom: 0;

      &:focus {
        border: none;
        box-shadow: none;
        background: transparent;
        border-radius: 0;
        transition: none;
      }
    }

    .Select-input {
      height: 28px;

      input {
        font-size: 15px;
        padding: 0;
      }
    }

    .Select-control {
      height: $panel-space-30;
      border-radius: $corner-radius;
    }

    .Select-menu-outer {
      font-weight: 500;
      font-size: 13px;
    }

    .Select--multi,
    .Select-value {
      background: $tc-gray-neutral-dark !important;
      border: none !important;
      font-family: 'Roboto';
      font-weight: 700;
      font-size: 10px;
      color: $tc-gray-80 !important;
      line-height: $panel-space-10 + 2;
      margin-top: $base-unit - 1;
      border-radius: $corner-radius + 1;
      position: relative;

      .Select-value-label {
        padding: $base-unit - 1 $panel-space-25 $base-unit - 1 $panel-space-7;
      }

      .Select-value-icon {
        position: absolute;
        right: 0;
        top: 4px;
        padding: 0 $panel-space-7;
        font-size: 14px;
        border: none;
        border-radius: 0 $corner-radius + 1 $corner-radius + 1 0;
      }
    }
  }

  &.track {
    :global {
      .Select-value {
        background: $tc-light-blue !important;
        color: $tc-white !important;
      }
    }
  }
}
