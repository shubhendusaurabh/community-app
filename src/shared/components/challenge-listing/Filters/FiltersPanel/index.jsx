/* eslint jsx-a11y/no-static-element-interactions:0 */

/**
 * Challenge filters panel.
 *
 * It contains:
 *  - Challenge keywords filter;
 *  - Challenge tracks filter;
 *  - Challenge dates range filter;
 *  - Clear and save filter buttons.
 *
 * Challenge keywords and tracks filters allow to choose multiple keywords from
 * the predefined sets, which should be passed into the component as string arrays
 * via the 'validKeywords' and 'validTracks' properties. The whole filters panel
 * can be hidden/displayed by setting the boolean 'hidden' property.
 *
 * Each time the user modifies any filter, this component triggers the callback
 * provided via the 'onFilter' property, if any, passing in the current filter
 * object.
 */

import _ from 'lodash';
import * as Filter from 'utils/challenge-listing/filter';
import React from 'react';
import PT from 'prop-types';
import Select from 'components/Select';
import moment from 'moment';
import UiSimpleRemove from '../../Icons/ui-simple-remove.svg';

import './style.scss';
import DateRangePicker from '../DateRangePicker';

export default function FiltersPanel({
  communityFilters,
  filterState,
  hidden,
  onClose,
  onSaveFilter,
  selectCommunity,
  selectedCommunityId,
  setFilterState,
  setSearchText,
  validKeywords,
  validSubtracks,
}) {
  let className = 'FiltersPanel';
  if (hidden) className += ' hidden';

  const communityOps = communityFilters.map(item => ({
    label: item.communityName,
    value: item.communityId,
  }));

  const mapOps = item => ({ label: item, value: item });
  const mapSubtracks = item => ({ label: item.name, value: item.subTrack });
  return (
    <div styleName={className}>
      <div styleName="header">
        <span styleName="title">Filters</span>
        <span styleName="close-icon" onClick={() => onClose()}>
          <UiSimpleRemove className="cross" />
        </span>
      </div>
      <div styleName="filters inGroup">
        <div styleName="filter-row">
          <div styleName="filter keywords">
            <label htmlFor="keyword-select" styleName="left-label">Keywords</label>
            <Select
              id="keyword-select"
              multi
              onChange={(value) => {
                const tags = value ? value.split(',') : undefined;
                setFilterState(Filter.setTags(filterState, tags));
              }}
              options={validKeywords.map(mapOps)}
              simpleValue
              value={filterState.tags ? filterState.tags.join(',') : null}
            />
          </div>
          <div styleName="filter community">
            <label htmlFor="community-select">Sub community</label>
            <Select
              autoBlur
              clearable={false}
              id="community-select"
              onChange={selectCommunity}
              options={communityOps}
              simpleValue
              value={selectedCommunityId}
            />
          </div>
        </div>
        <div styleName="filter-row">
          <div styleName="filter track">
            <label htmlFor="track-select" styleName="left-label">Subtrack</label>
            <Select
              id="track-select"
              multi
              onChange={(value) => {
                const subtracks = value ? value.split(',') : undefined;
                setFilterState(Filter.setSubtracks(filterState, subtracks));
              }}
              options={validSubtracks.map(mapSubtracks)}
              simpleValue
              value={
                filterState.subtracks ? filterState.subtracks.join(',') : null
              }
            />
          </div>
          <div styleName="filter dates">
            <label htmlFor="date-range-picker">Date range</label>
            <DateRangePicker
              endDate={filterState.endDate && moment(filterState.endDate)}
              id="date-range-picker"
              onDatesChange={(dates) => {
                let d = dates.endDate ? dates.endDate.toISOString() : null;
                let state = Filter.setEndDate(filterState, d);
                d = dates.startDate ? dates.startDate.toISOString() : null;
                state = Filter.setStartDate(state, d);
                setFilterState(state);
              }}
              startDate={
                filterState.startDate && moment(filterState.startDate)
              }
            />
          </div>
        </div>
      </div>
      <div styleName="buttons">
        <button
          styleName="white"
          className="tc-outline-btn"
          disabled={_.isEmpty(filterState)}
          onClick={() => {
            setFilterState({});
            setSearchText('');
          }}
        >Clear filters</button>
        <button
          styleName="blue"
          className="tc-blue-btn"
          disabled={_.isEmpty(filterState)}
          onClick={onSaveFilter}
        >Save filter</button>
      </div>
    </div>
  );
}

FiltersPanel.defaultProps = {
  hidden: false,
  onSaveFilter: _.noop,
  onClose: _.noop,
};

FiltersPanel.propTypes = {
  communityFilters: PT.arrayOf(PT.shape({
    communityId: PT.string.isRequired,
    communityName: PT.string.isRequired,
  })).isRequired,
  filterState: PT.shape().isRequired,
  hidden: PT.bool,
  onSaveFilter: PT.func,
  selectCommunity: PT.func.isRequired,
  selectedCommunityId: PT.string.isRequired,
  setFilterState: PT.func.isRequired,
  setSearchText: PT.func.isRequired,
  validKeywords: PT.arrayOf(PT.string).isRequired,
  validSubtracks: PT.arrayOf(PT.shape()).isRequired,
  onClose: PT.func,
};
