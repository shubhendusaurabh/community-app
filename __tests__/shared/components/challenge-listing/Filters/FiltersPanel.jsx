import React from 'react';
import _ from 'lodash';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import FiltersPanel from 'components/challenge-listing/Filters/FiltersPanel';

const onSaveFilter = jest.fn();
const selectCommunity = jest.fn();
const setFilterState = jest.fn();
const setSearchText = jest.fn();
const onClose = jest.fn();

const mockDatas = [{
  communityFilters: [
    {
      communityId: '123',
      communityName: 'Sample community',
    },
  ],
  filterState: {},
  hidden: true,
  onSaveFilter,
  selectCommunity,
  selectedCommunityId: '1',
  setFilterState,
  setSearchText,
  validKeywords: ['key', 'word'],
  validSubtracks: [{ name: 'sub', subTrack: 'sub' }, { name: 'track', subTrack: 'track' }],
  onClose,
}, {
  communityFilters: [
    {
      communityId: '123',
      communityName: 'Sample community',
    },
  ],
  filterState: {},
  hidden: false,
  onSaveFilter,
  selectCommunity,
  selectedCommunityId: '1',
  setFilterState,
  setSearchText,
  validKeywords: [],
  validSubtracks: [],
  onClose,
}];

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  _.forEach(mockDatas, (data) => {
    renderer.render((
      <FiltersPanel {...data} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

class Wrapper extends React.Component {
  componentDidMount() {}

  render() {
    return <FiltersPanel {...this.props} />;
  }
}

test('handle events', () => {
  const instance = TU.renderIntoDocument((<Wrapper {...mockDatas[0]} />));
  const matches = TU.scryRenderedDOMComponentsWithTag(instance, 'button');
  expect(matches).toHaveLength(2);

  TU.Simulate.click(matches[0]);
  expect(setFilterState).toHaveBeenCalledTimes(0);
  expect(setSearchText).toHaveBeenCalledTimes(0);

  TU.Simulate.click(matches[1]);
  expect(onSaveFilter).toHaveBeenCalledTimes(0);
});
