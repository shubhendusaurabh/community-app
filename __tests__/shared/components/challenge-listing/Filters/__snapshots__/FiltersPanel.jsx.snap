// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__FiltersPanel___1XPn6 src-shared-components-challenge-listing-Filters-FiltersPanel-___style__hidden___JoKmL"
>
  <div
    className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__header___dbqVu"
  >
    <span
      className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__title___n4gdC"
    >
      Filters
    </span>
    <span
      className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__close-icon___2EeEO"
      onClick={[Function]}
    >
      <UiSimpleRemove
        className="cross"
      />
    </span>
  </div>
  <div
    className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filters___3um3W src-shared-components-challenge-listing-Filters-FiltersPanel-___style__inGroup___3zhaT"
  >
    <div
      className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter-row___2Vfd_"
    >
      <div
        className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter___1-QUz src-shared-components-challenge-listing-Filters-FiltersPanel-___style__keywords___3Os3o"
      >
        <label
          className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__left-label___15kRu"
          htmlFor="keyword-select"
        >
          Keywords
        </label>
        <Select
          id="keyword-select"
          multi={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "key",
                "value": "key",
              },
              Object {
                "label": "word",
                "value": "word",
              },
            ]
          }
          simpleValue={true}
          value={null}
        />
      </div>
      <div
        className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter___1-QUz src-shared-components-challenge-listing-Filters-FiltersPanel-___style__community___3EFG7"
      >
        <label
          htmlFor="community-select"
        >
          Sub community
        </label>
        <Select
          autoBlur={true}
          clearable={false}
          id="community-select"
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Sample community",
                "value": "123",
              },
            ]
          }
          simpleValue={true}
          value="1"
        />
      </div>
    </div>
    <div
      className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter-row___2Vfd_"
    >
      <div
        className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter___1-QUz src-shared-components-challenge-listing-Filters-FiltersPanel-___style__track___1_Z_F"
      >
        <label
          className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__left-label___15kRu"
          htmlFor="track-select"
        >
          Subtrack
        </label>
        <Select
          id="track-select"
          multi={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "sub",
                "value": "sub",
              },
              Object {
                "label": "track",
                "value": "track",
              },
            ]
          }
          simpleValue={true}
          value={null}
        />
      </div>
      <div
        className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter___1-QUz src-shared-components-challenge-listing-Filters-FiltersPanel-___style__dates___3kH3c"
      >
        <label
          htmlFor="date-range-picker"
        >
          Date range
        </label>
        <DateRangePicker
          endDate={undefined}
          id="date-range-picker"
          onDatesChange={[Function]}
          startDate={undefined}
        />
      </div>
    </div>
  </div>
  <div
    className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__buttons___2r3xW"
  >
    <button
      className="tc-outline-btn src-shared-components-challenge-listing-Filters-FiltersPanel-___style__white___AWGFv"
      disabled={true}
      onClick={[Function]}
    >
      Clear filters
    </button>
    <button
      className="tc-blue-btn src-shared-components-challenge-listing-Filters-FiltersPanel-___style__blue___3L4J6"
      disabled={true}
      onClick={[Function]}
    >
      Save filter
    </button>
  </div>
</div>
`;

exports[`Matches shallow shapshot 2`] = `
<div
  className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__FiltersPanel___1XPn6"
>
  <div
    className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__header___dbqVu"
  >
    <span
      className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__title___n4gdC"
    >
      Filters
    </span>
    <span
      className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__close-icon___2EeEO"
      onClick={[Function]}
    >
      <UiSimpleRemove
        className="cross"
      />
    </span>
  </div>
  <div
    className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filters___3um3W src-shared-components-challenge-listing-Filters-FiltersPanel-___style__inGroup___3zhaT"
  >
    <div
      className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter-row___2Vfd_"
    >
      <div
        className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter___1-QUz src-shared-components-challenge-listing-Filters-FiltersPanel-___style__keywords___3Os3o"
      >
        <label
          className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__left-label___15kRu"
          htmlFor="keyword-select"
        >
          Keywords
        </label>
        <Select
          id="keyword-select"
          multi={true}
          onChange={[Function]}
          options={Array []}
          simpleValue={true}
          value={null}
        />
      </div>
      <div
        className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter___1-QUz src-shared-components-challenge-listing-Filters-FiltersPanel-___style__community___3EFG7"
      >
        <label
          htmlFor="community-select"
        >
          Sub community
        </label>
        <Select
          autoBlur={true}
          clearable={false}
          id="community-select"
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Sample community",
                "value": "123",
              },
            ]
          }
          simpleValue={true}
          value="1"
        />
      </div>
    </div>
    <div
      className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter-row___2Vfd_"
    >
      <div
        className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter___1-QUz src-shared-components-challenge-listing-Filters-FiltersPanel-___style__track___1_Z_F"
      >
        <label
          className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__left-label___15kRu"
          htmlFor="track-select"
        >
          Subtrack
        </label>
        <Select
          id="track-select"
          multi={true}
          onChange={[Function]}
          options={Array []}
          simpleValue={true}
          value={null}
        />
      </div>
      <div
        className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__filter___1-QUz src-shared-components-challenge-listing-Filters-FiltersPanel-___style__dates___3kH3c"
      >
        <label
          htmlFor="date-range-picker"
        >
          Date range
        </label>
        <DateRangePicker
          endDate={undefined}
          id="date-range-picker"
          onDatesChange={[Function]}
          startDate={undefined}
        />
      </div>
    </div>
  </div>
  <div
    className="src-shared-components-challenge-listing-Filters-FiltersPanel-___style__buttons___2r3xW"
  >
    <button
      className="tc-outline-btn src-shared-components-challenge-listing-Filters-FiltersPanel-___style__white___AWGFv"
      disabled={true}
      onClick={[Function]}
    >
      Clear filters
    </button>
    <button
      className="tc-blue-btn src-shared-components-challenge-listing-Filters-FiltersPanel-___style__blue___3L4J6"
      disabled={true}
      onClick={[Function]}
    >
      Save filter
    </button>
  </div>
</div>
`;
